src/xunit.assert/Asserts/Sdk/AssertEqualityComparer.cs

```diff
@@ -149,6 +158,36 @@ public bool Equals(T x, T y)
             return dictionaryYKeys.Count == 0;
         }
 
+        bool? CheckIfSetsAreEqual(T x, T y, TypeInfo typeInfo)
+        {
+            if (!IsSet(typeInfo))
+                return null;
+
+            var enumX = x as IEnumerable;
+            var enumY = y as IEnumerable;
+            if (enumX == null || enumY == null)
+                return null;
+
+            var objectEnumX = enumX.Cast<object>();
+            var objectEnumY = enumY.Cast<object>();
+            if (objectEnumX.Count() != objectEnumY.Count())
+                return false;
+
+            var setX = new HashSet<object>(objectEnumX, objectComparer);
```

Comment:
Found a way!
```cs
var elementType = typeof(T).GenericTypeArguments[0];
MethodInfo method = GetType().GetTypeInfo().GetDeclaredMethod("CompareTypedSets");
method = method.MakeGenericMethod(new Type[] { elementType });
return (bool)method.Invoke(this, new object[] { enumX, enumY });
```
where CompareTypedSets is:
```cs
bool CompareTypedSets<R>(IEnumerable enumX, IEnumerable enumY)
{
    var setX = new HashSet<R>(enumX.Cast<R>());
    var setY = new HashSet<R>(enumY.Cast<R>());
    return setX.SetEquals(setY);
}
```
I'll rewrite the commit history to include this. Thanks for the help!